js/classes/entities/foreground/weather_system.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * WeatherSystem manages what weather to spawn into a zone.
+ * 
+ * @author Caleb Krauter
+ */
+class WeatherSystem {
+
+    /**
+     * Pass in the type of weather, intensity and time of day.
```

Comment:
Makes sense. What would you typically jsDoc for a constructor?

Reply:
That's a good question. Yes I think so. It is useful for folks to understand what the parameter requirements are.

Reply:
I intended to ask what you would describe here in this jsDoc for a constructor instead of mentioning the parameters. I understand that a Doc should describe what a function or method is doing. Not sure what would make sense to describe for a constructor such as this.

Reply:
My point here is that you seem to be describing the parameters twice. You do like a comment about the parameters, then later in this, jsdoc you actually use the @param notation.

You only need to do the second one. Does that make sense?